boot/bootutil/src/bootutil_priv.h

```diff
@@ -143,8 +143,10 @@ _Static_assert(BOOT_IMAGE_NUMBER > 0, "Invalid value for BOOT_IMAGE_NUMBER");
 #else
 #define ARE_SLOTS_EQUIVALENT()    1
 
+#ifdef MCUBOOT_DIRECT_XIP
```

Comment:
I wonder if `#if defined(MCUBOOT_DIRECT_XIP) && defined(MCUBOOT_ENC_IMAGES)` might make this a little clearer.